src/blocks/weather.rs

```diff
@@ -140,42 +190,25 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
         .new_widget()
         .with_format(config.format.with_default("$weather $temp")?);
 
+    // met_no requires a one-time fetch of a set of "legends", which are
+    // translations of symbols into weather strings.
+    let legends_store: met_no::LegendsStore = match config.service {
+        WeatherService::MetNo(_) => met_no::get_legend().await?,
+        _ => HashMap::new(),
+    };
+
     loop {
+        let location = match config.autolocate {
+            true => find_ip_location().await.unwrap_or(None),
```

Comment:
This should be executed via `api.recoverable()`

Reply:
Fixed, see 7aa0069.